cpp/src/fil/infer.cu

```diff
@@ -28,7 +28,7 @@ struct vec {
   T data[N];
   inline __host__ __device__ vec() {
 #pragma unroll
-    for (int i = 0; i < N; ++i) data[i] = 0;
+    for (int i = 0; i < N; ++i) data[i] = T();
```

Comment:
Does this initialize to 0 for types like `int` or `float`?

Reply:
yes. see https://en.cppreference.com/w/cpp/language/value_initialization , line "4) otherwise, the object is zero-initialized."